Jenkinsfile

```diff
@@ -6,46 +6,63 @@
 // all possible error combinations.
 // See also the top comment in syncenginetestutils.h
 //
+// We are building "Linux - GCC" with "make" and "Linux - Clang" with ninja,
+// the combinations are more or less arbitrarily chosen. We just want to
+// check that both compilers and both CMake generators work. It's
+// unlikely that a specific generator only breaks with a specific
+// compiler.
+
+
+def linux = docker.image('dominikschmidt/owncloud-client-ci-image:latest')
+def win32 = docker.image('guruz/docker-owncloud-client-win32:latest')
 
 node('CLIENT') {
     stage 'Checkout'
         checkout scm
         sh '''git submodule update --init'''
 
-    stage 'Qt5'
-        sh '''rm -rf build
-		mkdir build
-		cd build
-		cmake -DCMAKE_BUILD_TYPE="Debug" -DUNIT_TESTING=1 -DWITH_TESTING=1 -DCMAKE_PREFIX_PATH=/var/lib/jenkins/qt/5.6.2 ..
-		make -j4
-		ctest -V --output-on-failure'''
-
-    stage 'Qt5 - clang'
-        sh '''rm -rf build
-		mkdir build
-		cd build
-		cmake -DCMAKE_BUILD_TYPE="Debug" -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DUNIT_TESTING=1 -DWITH_TESTING=1 -DCMAKE_PREFIX_PATH=/var/lib/jenkins/qt/5.6.2 ..
-		make -j4
-		ctest -V --output-on-failure'''
+    stage 'Linux - GCC'
+        linux.pull()
+        linux.inside {
+            sh '''
+            export HOME="$(pwd)/home"
+            rm -rf build home
+            mkdir build
+            cd build
+            cmake -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_BUILD_TYPE="Debug" -DUNIT_TESTING=1 ..
+            make -j4
+            LC_ALL=C.UTF-8 ctest -V --output-on-failure
+            '''
+        }
 
+    stage 'Linux - Clang'
+        linux.pull()
```

Comment:
should the pull only be done once?

Reply:
`.pull()` should be more or less a noop usually, unless the tag is changed in between...

My intention was to make the stages completely independent, so one can be commited out (like you did with the clang stage when working on this thing initially). I realize it might not be the best way to do it - also having the pull in the compile stage makes the times less comparable. I think I would like to move the pulls into their own stages .. sounds good?